src/main/java/org/apache/maven/buildcache/CacheControllerImpl.java

```diff
@@ -173,11 +173,14 @@ public CacheResult findCachedBuild(
             LOGGER.info("Attempting to restore project {} from build cache", projectName);
 
             // remote build first
-            result = findCachedBuild(mojoExecutions, context);
+            if (cacheConfig.isRemoteCacheEnabled()) {
```

Comment:
Please write integration tests for this flag. In pr #34, I added Wiremock-based integration tests to verify the remote cache behavior. The same approach could be reused here. It will be good to see a test that checks that remote is disabled.

Reply:
1. For testing flag and xml setting combinations, a lightweight unit test seems prudent.  I went ahead and added a unit test for CacheConfigImpl.  I tested the initialization and the remote parameters only, but structured it in a way that it should easily accommodate tests around other configurations.  A couple integration tests make sense as well, but I thought i'd wait until PR #34 is merged to build off that approach. Tests are a great way to guide meaningful cleanups and refactors, and I feel I need some more time and context to do this at an integration level.

2. Remote configurations are now preconditioned on any meaningful encompassing configuration behind the CacheConfig interface.

Reply:
Thank you, I clicked it!